### DISCUSSION TITLE
Coral mpcie connected to pcie 2.0 x1?

### DISCUSSION THREAD

**Discussion by avvalex** [blakeblackshear/frigate#7378] [General]
Hi, someone can kindly confirm if the Coral M-PCIE version can work with a motherbard Asrock J5040 ITX, that has a PCIE 2.0 1x ?

https://www.asrock.com/mb/Intel/J5040-ITX/index.it.asp

The datasheet of the Coral says "PCIe Gen2 x1 interface"....

Thanks!!

**Comment by krzakx** [comment]
hey, I think I can help you as this is also my current research topic.
You can see that on Dual TPU Edge product page https://coral.ai/products/m2-accelerator-dual-edgetpu/ they do mention:
![image](https://github.com/blakeblackshear/frigate/assets/18627339/6527a7b5-8c8a-47da-bee6-984037256d0c)
so by this, we can very likely say that for single TPU Edge -> PCIe Gen2 with 1 lane (x1) is enought.

**Reply by avvalex** [reply]
So, if i'm not mistaken, chances are:
M2 connector has 1 lane: single TPU M2-E key is ok, dual TPU will work only one;
M2 connector has 2 lane:  single TPU M2-E key is ok, dual TPU will work both.
We only don't know in my MOBO has an M2 with one or two lanes.....
